python/cudf/cudf/core/column/column.py

```diff
@@ -1948,9 +1948,17 @@ def as_column(
                 raise TypeError(
                     f"Cannot convert a {inferred_dtype} of object type"
                 )
+            elif (
+                not nan_as_null
+                and inferred_dtype == "string"
+                and any(
+                    isinstance(x, float) and np.isnan(x) for x in arbitrary
+                )
+            ):
+                raise MixedTypeError(f"Cannot have NaN with {inferred_dtype}")
```

Comment:
Instead could you transform the condition below to:

```python
# we _only_ want to check for nan. Other pandas NA values are OK.
elif nan_as_null is False and (
    any(isinstance(x, (np.floating, float)) and np.isnan(x) for x in np.asarray(arbitrary)
    and inferred_dtype not in ("decimal", "empty")
):
```

And see if that fixes the original issue?

Reply:
Um, nope. There might be some more handling needed. I'm working on it..